diff --git a/cflib/crypt/botan/asn1_time.cpp b/cflib/crypt/botan/asn1_time.cpp
index b79e8a9..63b4b96 100644
--- a/cflib/crypt/botan/asn1_time.cpp
+++ b/cflib/crypt/botan/asn1_time.cpp
@@ -232,6 +232,9 @@ std::string X509_Time::readable_string() const
 
    std::string output(24, 0);
 
+   #if defined(_MSC_VER)
+   #pragma warning(disable:4996)
+   #endif
    std::sprintf(&output[0], "%04d/%02d/%02d %02d:%02d:%02d UTC",
                 year, month, day, hour, minute, second);
 
diff --git a/cflib/crypt/botan/internal/def_pk_ops.cpp b/cflib/crypt/botan/internal/def_pk_ops.cpp
index f941245..30fa9a2 100644
--- a/cflib/crypt/botan/internal/def_pk_ops.cpp
+++ b/cflib/crypt/botan/internal/def_pk_ops.cpp
@@ -140,7 +140,7 @@ Core_Engine::get_signature_op(const Private_Key& key, const std::string& emsa, R
    }
 
 PK_Ops::Verification*
-Core_Engine::get_verify_op(const Public_Key& key, const std::string& emsa, RandomNumberGenerator&) const
+Core_Engine::get_verify_op(const Public_Key& key, const std::string& /*emsa*/, RandomNumberGenerator&) const
    {
 #if defined(BOTAN_HAS_RSA)
    if(const RSA_PublicKey* s = dynamic_cast<const RSA_PublicKey*>(&key))
diff --git a/cflib/crypt/botan/internal/donna.cpp b/cflib/crypt/botan/internal/donna.cpp
index 29d40ce..611cd48 100644
--- a/cflib/crypt/botan/internal/donna.cpp
+++ b/cflib/crypt/botan/internal/donna.cpp
@@ -340,6 +340,9 @@ fmonty(limb *x2, limb *z2, /* output 2Q */
 static void
 swap_conditional(limb a[5], limb b[5], limb iswap) {
   unsigned i;
+  #if defined(_MSC_VER)
+  #pragma warning(suppress:4146)
+  #endif
   const limb swap = -iswap;
 
   for (i = 0; i < 5; ++i) {
diff --git a/cflib/crypt/botan/internal/msg_client_kex.cpp b/cflib/crypt/botan/internal/msg_client_kex.cpp
index c8dc2aa..b9e2cac 100644
--- a/cflib/crypt/botan/internal/msg_client_kex.cpp
+++ b/cflib/crypt/botan/internal/msg_client_kex.cpp
@@ -383,7 +383,7 @@ Client_Key_Exchange::Client_Key_Exchange(const std::vector<byte>& contents,
             else
                m_pre_master = shared_secret;
             }
-         catch(std::exception &e)
+         catch(std::exception &/*e*/)
             {
             /*
             * Something failed in the DH computation. To avoid possible
diff --git a/cflib/crypt/botan/internal/tls_handshake_io.h b/cflib/crypt/botan/internal/tls_handshake_io.h
index 34873c3..dc02f62 100644
--- a/cflib/crypt/botan/internal/tls_handshake_io.h
+++ b/cflib/crypt/botan/internal/tls_handshake_io.h
@@ -165,7 +165,7 @@ class Datagram_Handshake_IO : public Handshake_IO
          Message_Info(u16bit e, Handshake_Type mt, const std::vector<byte>& msg) :
             epoch(e), msg_type(mt), msg_bits(msg) {}
 
-         Message_Info(const Message_Info& other) = default;
+         Message_Info(const Message_Info& /*other*/) = default;
 
          Message_Info() : epoch(0xFFFF), msg_type(HANDSHAKE_NONE) {}
 
diff --git a/cflib/crypt/botan/internal/tls_handshake_state.cpp b/cflib/crypt/botan/internal/tls_handshake_state.cpp
index 1110870..85b6fb1 100644
--- a/cflib/crypt/botan/internal/tls_handshake_state.cpp
+++ b/cflib/crypt/botan/internal/tls_handshake_state.cpp
@@ -369,7 +369,7 @@ std::pair<std::string, Signature_Format>
 Handshake_State::understand_sig_format(const Public_Key& key,
                                        std::string hash_algo,
                                        std::string sig_algo,
-                                       bool for_client_auth) const
+                                       bool /*for_client_auth*/) const
    {
    const std::string algo_name = key.algo_name();
 
diff --git a/cflib/crypt/botan/internal/tls_record.cpp b/cflib/crypt/botan/internal/tls_record.cpp
index 3edeab7..bc9acc5 100644
--- a/cflib/crypt/botan/internal/tls_record.cpp
+++ b/cflib/crypt/botan/internal/tls_record.cpp
@@ -305,7 +305,7 @@ size_t fill_buffer_to(secure_vector<byte>& readbuf,
 *
 * @fixme This should run in constant time
 */
-size_t tls_padding_check(size_t block_size,
+size_t tls_padding_check(size_t /*block_size*/,
                          const byte record[],
                          size_t record_len)
    {
diff --git a/cflib/crypt/botan/poly1305.cpp b/cflib/crypt/botan/poly1305.cpp
index 8000ebb..2e6028f 100644
--- a/cflib/crypt/botan/poly1305.cpp
+++ b/cflib/crypt/botan/poly1305.cpp
@@ -17,7 +17,7 @@ void Poly1305::clear()
    m_buf_pos = 0;
    }
 
-void Poly1305::key_schedule(const byte key[], size_t key_len)
+void Poly1305::key_schedule(const byte key[], size_t /*key_len*/)
    {
    m_buf_pos = 0;
    m_buf.resize(16);
diff --git a/cflib/crypt/botan/x509_obj.cpp b/cflib/crypt/botan/x509_obj.cpp
index 746fc73..63adccc 100644
--- a/cflib/crypt/botan/x509_obj.cpp
+++ b/cflib/crypt/botan/x509_obj.cpp
@@ -199,7 +199,7 @@ bool X509_Object::check_signature(const Public_Key& pub_key) const
 
       return verifier.verify_message(tbs_data(), signature());
       }
-   catch(std::exception& e)
+   catch(std::exception& /*e*/)
       {
       return false;
       }
diff --git a/cflib/crypt/botan/x509path.cpp b/cflib/crypt/botan/x509path.cpp
index 111c4c3..1d55b1d 100644
--- a/cflib/crypt/botan/x509path.cpp
+++ b/cflib/crypt/botan/x509path.cpp
@@ -163,7 +163,7 @@ check_chain(const std::vector<X509_Certificate>& cert_path,
             else if(ocsp_status == Certificate_Status_Code::OCSP_RESPONSE_GOOD)
                continue;
             }
-         catch(std::exception& e)
+         catch(std::exception& /*e*/)
             {
             //std::cout << "OCSP error: " << e.what() << "\n";
             }
